src/test/java/com/marklogic/mule/extension/AbstractFlowTester.java

```diff
@@ -16,6 +18,13 @@ public abstract class AbstractFlowTester extends MuleArtifactFunctionalTestCase
     protected final static String TEXT_HELLO_WORLD = "Hello, World!\n";
     protected final static String XML_HELLO_WORLD = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Hello>World</Hello>";
 
+    @Before
+    public void prepareDatabase() throws Exception {
+        if ((getConfigFiles() != null) && Arrays.asList(getConfigFiles()).contains("prepare-database")) {
+            flowRunner("prepare-database").run();
```

Comment:
This looks good, but I think it can get a little simpler with less code:

First, we should always run this flow. There's never a reason not to run it if we're testing a flow, which a subclass of `AbstractFlowTester` is always doing.

Second - to make life easier for the subclasses and to keep them from having to care that "prepare-database" even exists:

1. Add a `protected abstract String getFlowTestFile()` method to this class (a better name than "getConfigFile()" IMO).
2. Have each subclass override that with their single test file. 
3. This class then implements Mule's `getConfigFiles()` and returns the result of `getFlowTestFile()` and "prepare-database-flow.xml". 

Now subclasses don't care about "prepare-database" at all. And when we create a new test class that extends AFT, it's immediately forced to define its flow test file.